### PR TITLE
Refactor Finger letter print endpoint

### PR DESCRIPTION
**Pull request by peggy-quartech**
# Description

This PR includes the following proposed change(s):

- Refactor Finger letter print endpoint, get eventId from event repo, then update result in event table.

### CODE REVIEW

File: src/Spd.Manager.Printing/PrintingManager.cs [hunk 1]
```diff
@@ -64,10 +80,32 @@ private static DocumentTransformRequest CreateDocumentTransformRequest(PrintJob
         };
     }
 
-    private async Task<string> PrintViaBcMailPlus(BcMailPlusTransformResponse bcmailplusResponse, CancellationToken cancellationToken)
+    private async Task<string> PrintViaBcMailPlus(BcMailPlusTransformResponse bcmailplusResponse, Guid eventId, CancellationToken cancellationToken)
```
Comment (ruaraya) [inline src/Spd.Manager.Printing/PrintingManager.cs:83]: The name of this method and what it is doing (update) is a bit misleading. Should we change the name?

File: src/Spd.Resource.Repository/Event/Contract.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+using MediatR;
+
+namespace Spd.Resource.Repository.Event
+{
+    public interface IEventRepository
+    {
+        public Task<EventResp> GetAsync(Guid eventId, CancellationToken cancellationToken);
+        public Task<Unit> ManageAsync(EventUpdateCmd cmd, CancellationToken ct);
+    }
+
```
Comment (ruaraya) [inline src/Spd.Resource.Repository/Event/Contract.cs:10]: Please remove extra line break

File: src/Spd.Resource.Repository/Event/EventRepository.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        spd_eventqueue? eventQueue = await _context.GetEventById(cmd.Id, ct);
+        _mapper.Map<EventUpdateCmd, spd_eventqueue>(cmd, eventQueue);
+        _context.UpdateObject(eventQueue);
+        await _context.SaveChangesAsync(ct);
+        return default;
+    }
+}
+
+
```
Comment (ruaraya) [inline src/Spd.Resource.Repository/Event/EventRepository.cs:34]: Please remove extra line break

File: src/Spd.Resource.Repository/Event/Contract.cs (hunk 2 above)
Comment (peggy-quartech) [reply src/Spd.Resource.Repository/Event/Contract.cs:10]: sure

File: src/Spd.Resource.Repository/Event/EventRepository.cs (hunk 3 above)
Comment (peggy-quartech) [reply src/Spd.Resource.Repository/Event/EventRepository.cs:34]: sure

File: src/Spd.Manager.Printing/PrintingManager.cs (hunk 1 above)
Comment (peggy-quartech) [reply src/Spd.Manager.Printing/PrintingManager.cs]: add more error processing and refactor the code.

**Review by ruaraya — APPROVED** [review]

**Pull request merged**